build-tools/scripts/JdkDirectoryFinder.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+public class JdkDirectoryFinder : Task {
+
+	[Required]
+	public string JdksRoot { get; set; }
+
+	public string MaximumJdkVersion { get; set; }
+
+	[Output]
+	public string PreferredJdkRoot { get; set; }
+
+	static Regex VersionExtractor = new Regex (@"(?<version>[\d]+(\.\d+)+)");
```

Comment:
make it faster by passing the 2nd parameter to the constructor: `, RegexOptions.Compiled`